### PR TITLE
[Feature] [Domain] Describe adding comment by admin

### PR DESCRIPTION
**Pull request by lchrusciel**

### CODE REVIEW

File: features/commenting_an_order_by_an_administrator.feature [hunk 1]
```diff
@@ -0,0 +1,36 @@
+@administrator_order_comments
+Feature: Commenting an order by an administrator
+    In order to respond to the customers comment
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: `In order to communicate with the customer that made an order`

File: features/commenting_an_order_by_an_administrator.feature [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        And the store allows paying with "Cash on Delivery"
+        And there was account of "john.doe@gmail.com" with password "sylius"
+        And there is a customer "john.doe@gmail.com" that placed an order "#00000022"
+        And the customer bought a single "PHP T-Shirt"
+        And the customer chose "Free" shipping method to "United States" with "Cash on Delivery" payment
+        And I am logged in as an administrator
+
+    @domain
+    Scenario: Administrator commented an order
+        When I comment an order "#00000022" with "How can I help you?"
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: the order

File: features/commenting_an_order_by_an_administrator.feature [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        Then this order should have a comment with "How can I help you?" from this customer
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: customer? ;>

File: features/commenting_an_order_by_an_administrator.feature [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    @domain
+    Scenario: Administrator cannot comment an order with an empty message
+        When I try to comment an order "#00000022" with an empty message
+        Then this order should not have empty comment from this customer
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: `Then this order should not have any comments`
`And I should be notified that a message is required`

File: features/commenting_an_order_by_an_administrator.feature [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    @todo
+    Scenario: Administrator cannot comment an order which does not exist
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: This is the weirdest scenario I've seen. How can you even try commenting an order that does not exist? You cannot go to order show of a not existing order xD

File: features/commenting_an_order_by_an_administrator.feature [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        When I try to comment a not existing order with "How can I help you?"
+        Then this order should not have empty comment from this customer
+
+    @todo
+    Scenario: Sending the email notification to the customer about unread administrator's comments
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: an email notification
about unread <del>administrator's</del> comments

File: features/commenting_an_order_by_an_administrator.feature [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        Given I have commented the order "#00000022" with "How can I help you?"
+        Then the notification email should be sent to the administrator
```
Comment (CoderMaggie) [inline features/commenting_an_order_by_an_administrator.feature]: to the customer ;>

**Review by CoderMaggie — CHANGES_REQUESTED** [review]

File: tests/Comments/Domain/Model/CommentTest.php [hunk 8]
```diff
@@ -42,7 +56,31 @@ public function it_cannot_be_created_with_empty_message(): void
...
+
+        Comment::orderByAdministrator($order, 'test@test.com', '');
+    }
+
+    /**
+     * @test
+     *
+     * @expectedException \DomainException
+     */
+    public function customer_comment_cannot_be_created_with_invalid_rmail(): void
```
Comment (CoderMaggie) [inline tests/Comments/Domain/Model/CommentTest.php]: rmail!

File: tests/Behat/Context/Domain/AdministratorOrderCommentsContext.php [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                'There are no order comment with the "%s" message for the "%s" order from the "%s" customer',
+                $message, $order->getNumber(), $user->getEmail()
+            ));
+        }
+    }
+
+    /**
+     * @Then I should be notified that comment is invalid
+     */
+    public function thisOrderShouldNotHaveEmptyCommentFromThisCustomer(): void
```
Comment (CoderMaggie) [inline tests/Behat/Context/Domain/AdministratorOrderCommentsContext.php]: the function name should also change I suppose ;>

File: tests/Behat/Context/Domain/AdministratorOrderCommentsContext.php [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    {
+        /** @var AdminUserInterface $user */
+        $user = $this->sharedStorage->get('administrator');
+        $this->sharedStorage->set('comment', Comment::orderByAdministrator($order, $user->getEmail(), $message));
+    }
+
+    /**
+     * @When I try to comment the order :order with an empty message
+     */
+    public function aCustomerTryToCommentsTheOrderWithEmptyMessage(OrderInterface $order): void
```
Comment (CoderMaggie) [inline tests/Behat/Context/Domain/AdministratorOrderCommentsContext.php]: `aCustomerTriesToCommentTheOrderWithEmptyMessage`

File: src/Domain/Model/Comment.php [hunk 11]
```diff
@@ -51,6 +52,21 @@ public static function orderByCustomer(OrderInterface $order, string $customerEm
         return $comment;
     }
 
+    public static function orderByAdministrator(OrderInterface $order, string $customerEmail, string $message): self
```
Comment (Arminek) [inline src/Domain/Model/Comment.php]: `$customerEmail` -> `$administratorEmail`

**Review by CoderMaggie — APPROVED** [review]

**Pull request merged**